src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Formatting/RazorFormattingService.cs

```diff
@@ -240,4 +242,44 @@ private static void UnwrapCSharpSnippets(TextEdit[] snippetEdits)
             snippetEdit.NewText = unwrappedText;
         }
     }
+
+    // This method counts the occurrences of CRLF and LF line endings in the original text. If LF line endings are more
+    // prevalent, it removes any standalone CR characters from the text edits to ensure consistency with the LF style.
+    // This can be removed once we figure out how to get proper line ending information from the client.
+    private TextEdit[] NormalizeLineEndings(SourceText originalText, TextEdit[] minimalEdits)
+    {
+        var (crlfCount, lfCount) = CountLineEndings(originalText);
+        if (lfCount > crlfCount)
+        {
+            minimalEdits = minimalEdits
+                .Where(edit => edit.NewText != "\r")
```

Comment:
I'm surprised this works, but I wonder if it means we don't need to count line endings at all. I can't think of a scenario where an entire edit would be solely a `\r`, except in situations where our edits are changing line endings.

Seems like maybe by hardcoding our system to always use CRLF we might have lucked into an easy fix?

The other thing I think would be interesting to try is to call `CountLineEndings` in the constructor of `FormattingContext` and have it set `FormattingContext.NewLineString` to the right line ending. If we do that, does it negate the need for thsi post-formatting cleanup?

Reply:
Might be worth adding an assert after this to ensure that none of the edits even contain an `\r`. Trying to think of a scenario where formatting adds the newline as part of an edit, rather than in an entirely separate edit, and making sure we catch that.

Reply:
After adding some tests, I actually need to change this implementation. Allow me to clarify a few things:

1. I tested manually changing the value of `FormattingContext.NewLineString` and it didn't change anything in regards to the CSharp formatting (I think other formatting service actually read and use that field). I believe that Roslyn or whatever service is actually doing the formatting detects what line-ending is used in the `csharpSourceText` file, and apply the final edits based on that. And I also believe that the `csharpSourceText` file uses the system default line ending, which will explain why this bug is not reproducible on Mac. 

2. Because of 1. When the formatter wants to add a new line to a file with LF line ending, it will add a `\r\n`. In which case we will have to take just the `\r` out. Hence the changes in my latest commit. 

3. "Seems like maybe by hardcoding our system to always use CRLF we might have lucked into an easy fix?" -> I strongly believe and hope so.